lib/utils/fileSorting.ts

```diff
@@ -0,0 +1,77 @@
+import type { Node } from '../files/node'
+import { orderBy } from './sorting'
+
+export enum FilesSortingMode {
+	Name = 'basename',
+	Modified = 'mtime',
+	Size = 'size',
+}
+
+export interface FilesSortingOptions {
+	/**
+	 * They key to order the files by
+	 * @default FilesSortingMode.Name
+	 */
+	sortingMode?: FilesSortingMode
```

Comment:
That means we cannot sort by arbitrary values?

Reply:
Should we allow it? I just wanted to provide a clean interface, but we can of course also accept other values here if you like :)

Reply:
Ah, I confused myself.
I was thiking about custom columns, but I apparently had planned this already.

You have to provide a sort function if you want your custom column to be sortable, so this piece of code is not relevant, you're right!

https://github.com/nextcloud/server/blob/a8f4b25ae834778e120ea011a3c98aa77ef90c51/apps/files/src/views/FilesList.vue#L311-L315
https://github.com/nextcloud/server/blob/97cd038cf20c5015d9dfecd0e9283367391358d2/apps/files/src/components/FilesListTableHeader.vue#L61-L69

Reply:
If you ignore types you can even pass every valid property of the Node class^^